applications/FluidDynamicsApplication/custom_processes/apply_mach_dependent_boundary_conditions.cpp

```diff
@@ -0,0 +1,296 @@
+
+    KRATOS_CATCH("")
+}
+
+
+void ApplyMachDependentBoundaryConditions::BoundaryConditionUtility::
+    ActivateIfInsideTimeInterval(const double time)
+{
+    if(mInterval.IsInInterval(time))
+    {
+        mEnforceInternal = &FixDof;
+    }
+    else
+    {
+        mEnforceInternal = &DoNothing;
+    }
+}
+
+
+void ApplyMachDependentBoundaryConditions::BoundaryConditionUtility::
+    Enforce(NodeType & rNode) const
+{
+    mEnforceInternal(*this, rNode);
+}
+
+
+void ApplyMachDependentBoundaryConditions::BoundaryConditionUtility::
+    Free(NodeType & rNode) const
+{
+    rNode.pGetDof(*mpVariable)->FreeDof();
+}
+
+
+void ApplyMachDependentBoundaryConditions::BoundaryConditionUtility::FixDof(
+    const BoundaryConditionUtility & rUtility,
+    NodeType & rNode)
+{
+    rNode.GetSolutionStepValue(*rUtility.mpVariable) = rUtility.mValue;
+    rNode.pGetDof(*rUtility.mpVariable)->FixDof();
+}
+
+
+void ApplyMachDependentBoundaryConditions::BoundaryConditionUtility::DoNothing(
+    const BoundaryConditionUtility & rUtility,
+    NodeType & rNode)
+{
+}
+
+
+void ApplyMachDependentBoundaryConditions::ReadBoundaryCondition(
+    std::vector<BoundaryConditionUtility> & rBCList,
+    Parameters Parameters)
+{
+    KRATOS_TRY
+
+    KRATOS_ERROR_IF_NOT(Parameters.Has("variable_name"))
+        << "Missing string field 'variable_name' in boundary condition:\n" << Parameters  << std::endl;
+    
+    KRATOS_ERROR_IF_NOT(Parameters.Has("value"))
+        << "Missing double (or vector) field 'value' in boundary condition:\n" << Parameters  << std::endl;
+
+    // Reading interval
+    IntervalUtility interval_utility{Parameters};
+    
+    // Reading value and acting depending on if it's vector or double
+    if(Parameters["value"].IsDouble())
+    {
+        rBCList.emplace_back(
+            Parameters["variable_name"].GetString(),
+            Parameters["value"].GetDouble(),
+            interval_utility
+        );
+        return;
+    }
+
+    if(Parameters["value"].IsVector())
+    {
+        KRATOS_ERROR_IF_NOT(Parameters.Has("constrained"))
+            << "Missing boolean array field 'constrained' in boundary condition:\n"
+            << Parameters  << std::endl;
+
+        const auto values = Parameters["value"].GetVector();
+        const auto constraints = Parameters["constrained"];
+
+        const std::size_t n_components = values.size();
+        const std::size_t n_constraints = constraints.size();
+
+        KRATOS_ERROR_IF(n_constraints != n_components)
+            << "The number of values specified must be the same as number of constraints specified:\n"
+            << Parameters << std::endl;
+
+        KRATOS_ERROR_IF(n_components > 3)
+            << "Allowed vector variables are at most 3-dimensional:\n"
+            << Parameters << std::endl;
+
+        std::string variable_component_name = Parameters["variable_name"].GetString() + "_X";
+        
+        for(std::size_t i=0; i<n_components; i++)
```

Comment:
```suggestion
        for(std::size_t i=0; i<n_components; ++i)
```
Minor but this avoids a temporary copy of `i`. Hence, if not strictly needed to use `i++`, the `++i` version is preferred.

Reply:
The compiler makes these equivalent even as low as `-O1`:
i++: https://godbolt.org/z/ccohxKqxn
++i: https://godbolt.org/z/njd53TMG8

there is no copy, they generate the same assembly